client.go

```diff
@@ -35,6 +33,9 @@ type Config struct {
 
 // Client provides an interface for interacting with the S3 API.
 type Client struct {
+	// HTTPClient overrides the default http.Client.
+	HTTPClient *http.Client
```

Comment:
we can keep this client private. do you see any reason for keeping this public?

Reply:
I went back and forth on where to include it. Most cases it will not be used so I removed it from the `New(...)` function parameters. It didn't feel right in `Config{}` either but that might be the correct place. Functional options felt heavy. I was hoping to get your thoughts on it.

Reply:
although wasihttp modifies the http.DefaultClient to use transport provided by wasihttp sdk, I would like to keep this as an explicit option. 

maybe we can have a WithHTTPClient function like a lot of other sdk provides?

Reply:
Updated with your suggestion